src/Env/EnvFactory.php

```diff
@@ -27,8 +29,8 @@ class EnvFactory
      *
      * @return Env
      */
-    public static function createEnv(EnvEnum $type)
+    public static function createEnv(EnvEnum $type, AppVendor $appVendor, Dependency ...$dependencies)
```

Comment:
I agree with you, the signature may change and type hints should be consistent, I'll change to `array $dependencies` so we don't have to do it later.

However, Imho it makes sense to use variadics here, it should not have any particular cost to use them so even if they indeed solve some specific needs, I don't think it's wrong to use them for basic needs (i.e. do not have to iterate over all elements of an array for being sure they are of the good type).